From e9c07ed3f1c4dd3bf23373c16189b298307fb67c Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 02:16:43 -0400
Subject: [iommu] vt-d: Suppress compiler warnings

Message-id: <20140905021643.32574.64447.stgit@gir.stowe>
Patchwork-id: 91370
O-Subject: [RHEL7.1 PATCH 11/17] iommu/vt-d: Suppress compiler warnings
Bugzilla: 1136537
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1136537
Upstream Status: aa4d066a2a8041b7e73cee68ce5499aca29f265e

commit aa4d066a2a8041b7e73cee68ce5499aca29f265e
Author: Yijing Wang <wangyijing@huawei.com>
Date:   Mon May 26 20:14:06 2014 +0800

    iommu/vt-d: Suppress compiler warnings

    suppress compiler warnings:
    drivers/iommu/intel-iommu.c: In function 'device_to_iommu':
    drivers/iommu/intel-iommu.c:673: warning: 'segment' may be used uninitialized in this function
    drivers/iommu/intel-iommu.c: In function 'get_domain_for_dev.clone.3':
    drivers/iommu/intel-iommu.c:2217: warning: 'bridge_bus' may be used uninitialized in this function
    drivers/iommu/intel-iommu.c:2217: warning: 'bridge_devfn' may be used uninitialized in this function

    Signed-off-by: Yijing Wang <wangyijing@huawei.com>
    Signed-off-by: Joerg Roedel <jroedel@suse.de>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index 8263fee..c32c6e6 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -670,7 +670,7 @@ static struct intel_iommu *device_to_iommu(struct device *dev, u8 *bus, u8 *devf
  struct intel_iommu *iommu;
  struct device *tmp;
  struct pci_dev *ptmp, *pdev = NULL;
- u16 segment;
+ u16 segment = 0;
  int i;
 
  if (dev_is_pci(dev)) {
@@ -2216,7 +2216,8 @@ static struct dmar_domain *get_domain_for_dev(struct device *dev, int gaw)
  struct device_domain_info *info;
  struct pci_dev *dev_tmp = NULL;
  unsigned long flags;
- u8 bus, devfn, bridge_bus, bridge_devfn;
+ u8 bus, devfn;
+ u8 bridge_bus = 0, bridge_devfn = 0;
 
  domain = find_domain(dev);
  if (domain)
-- 
1.7.1